### PR TITLE
mu4e: add in code to handle flags (e.g. replied)

### LINKED ISSUE #25: Flags not being set after sending a message (mu4e)
Flags are not being set on a message after replying or forwarding when using mu4e. This is handled in mu4e by `mu4e-sent-handler` (see `mu4e-compose.el`) which is called after a message is sent. This will add the flags for replied/forwarded to the parent message if needed. I'm not sure how best to do this for org-msg, as that function in mu4e is called by it's process filter (see `mu4e-proc.el`), and I didn't quite understand how that works/how it should be used here. Maybe just call `mu4e-sent-handler` directly?

### PR DESCRIPTION
**Pull request by podiki**
This is to fix #25. This code is currently directly from `mu4e` without any edits to style for `org-msg`. This works for me, but would be good to have someone else test it as well, as I mostly compose in plain text so may not be thoroughly testd.

### CODE REVIEW

File: org-msg.el [hunk 1]
```diff
@@ -1075,9 +1075,34 @@ HTML emails."
...
+              (mu4e~compose-setup-fcc-maybe)
+              (widen)) nil t)
+  ;; when the message has been sent.
+  (add-hook 'message-sent-hook
+            (lambda () ;;  mu4e~compose-mark-after-sending
+              (setq mu4e-sent-func 'mu4e-sent-handler)
+              (mu4e~proc-sent (buffer-file-name))) nil t)
+  ;; mark these two hooks as permanent-local, so they'll survive mode-changes
+  ;;  (put 'mu4e~compose-save-before-sending 'permanent-local-hook t)
+  (put 'mu4e~compose-mark-after-sending 'permanent-local-hook t))
```
Comment (jeremy-compostella) [inline org-msg.el:1105]: Does this really serve any purpose ? `mu4e~compose-mark-after-sending` seems to not exist in mu4e => I don't think it make any difference.

I was probably useful when they were not using a lambda functions but with the new code I think this is just a useless instruction.

File: org-msg.el [hunk 2]
```diff
@@ -1075,9 +1075,34 @@ HTML emails."
...
-  (mu4e~compose-setup-fcc-maybe)
   (when mu4e-compose-complete-addresses
-    (mu4e~compose-setup-completion)))
+    (mu4e~compose-setup-completion))
+
+  ;; the following code is verbatim from mu4e-compse.el, mu4e-compose-mode
+  ;; this will setup fcc (saving sent messages) and handle flags
+  ;; (e.g. replied to)
+  ;;
+  ;; setup the fcc-stuff, if needed
```
Comment (jeremy-compostella) [inline org-msg.el:1085]: unnecessary with your introduction.

File: org-msg.el (hunk 1 above)
Comment (podiki) [reply org-msg.el:1105]: Looks like you are correct, I don't see that function in mu4e (anymore)

**Comment by podiki** [comment]
Thanks for the review, made those changes.

**Comment by jeremy-compostella** [comment]
Nice but could you just make it one patch ?

**Comment by podiki** [comment]
Ok, think I managed (never done much rebase/squashing but think I figured it out).

**Pull request merged**